File: belajarandroid/CrudUnitTesting/app/src/main/java/com/example/crudunittesting/adapter/NasabahAdapter.java
package com.example.crudunittesting.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.example.crudunittesting.R;
import com.example.crudunittesting.model.Nasabah;

import java.util.List;

public class NasabahAdapter extends BaseAdapter {
    Context context;
    private List<Nasabah> list;

    public NasabahAdapter(Context context,List<Nasabah> list){
        this.context=context;
        this.list=list;
    }

    @Override
    public int getCount() {
        return list.size();
    }

    @Override
    public Object getItem(int position) {
        return list.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if(convertView==null){
            LayoutInflater inflater=LayoutInflater.from(this.context);
            convertView=inflater.inflate(R.layout.item_nasabah,null);
        }

        Nasabah nasabah = list.get(position);
        TextView namaTextView=(TextView) convertView.findViewById(R.id.nama);
        TextView alamatTextView =(TextView) convertView.findViewById(R.id.alamat);

        namaTextView.setText(nasabah.getNama());
        alamatTextView.setText(nasabah.getAlamat());

        return convertView;
    }
}

File: belajarandroid/AndroidAdapter/app/src/main/java/com/example/androidadapter/adapter/nasabahadapter.kt
package com.example.androidadapter.adapter
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import com.example.androidadapter.model.Nasabah
import com.example.androidadapter.R

class nasabahadapter(var context: Context, private val list: List<Nasabah>) : BaseAdapter() {
    override fun getCount(): Int {
        return list.size
    }

    override fun getItem(position: Int): Any {
        return list[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View, parent: ViewGroup): View {
        var convertView = convertView
        if (convertView == null) {
            val inflater = LayoutInflater.from(context)
            convertView = inflater.inflate(R.layout.item_nasabah, null)
        }
        val nasabah = list[position]
        val namaTextView = convertView.findViewById<View>(R.id.nama) as TextView
        val lokasiTextView = convertView.findViewById<View>(R.id.lokasi) as TextView
        namaTextView.text = nasabah.nama
        lokasiTextView.text = nasabah.alamat
        return convertView
    }

}